src/libraries/System.Private.CoreLib/src/System/Collections/BitArray.cs

```diff
@@ -0,0 +1,972 @@
+            {
+                for (; i <= (uint)values.Length - Vector128<byte>.Count * 2u; i += (uint)Vector128<byte>.Count * 2u)
+                {
+                    Vector128<byte> lowerVector = Vector128.LoadUnsafe(ref value, i);
+                    Vector128<byte> lowerIsFalse = Vector128.Equals(lowerVector, Vector128<byte>.Zero);
+                    uint lowerResult = lowerIsFalse.ExtractMostSignificantBits();
+
+                    Vector128<byte> upperVector = Vector128.LoadUnsafe(ref value, i + (uint)Vector128<byte>.Count);
+                    Vector128<byte> upperIsFalse = Vector128.Equals(upperVector, Vector128<byte>.Zero);
+                    uint upperResult = upperIsFalse.ExtractMostSignificantBits();
+
+                    Unsafe.WriteUnaligned(
+                        ref Unsafe.Add(ref arrayRef, sizeof(uint) * (i / 32u)),
+                        ~((upperResult << 16) | lowerResult));
+                }
+            }
+
+        Remainder:
+            for (; i < (uint)values.Length; i++)
+            {
+                if (values[i])
+                {
+                    (uint byteIndex, uint bitOffset) = Math.DivRem(i, BitsPerByte);
+                    _array[byteIndex] |= (byte)(1 << (int)bitOffset);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BitArray"/> class that contains bit values
+        /// copied from the specified array of 32-bit integers.
+        /// </summary>
+        /// <param name="values">An array of integers containing the values to copy, where each integer represents 32 consecutive bits.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="values"/> is null.</exception>
+        /// <exception cref="ArgumentException">The length of <paramref name="values"/> in bits is greater than <see cref="int.MaxValue"/>.</exception>
+        /// <remarks>
+        /// The number in the first <paramref name="values"/> array element represents bits 0 through 31, the second number in the array represents
+        /// bits 32 through 63, and so on. The Least Significant Bit of each integer represents the lowest index value:
+        /// "<paramref name="values"/>[0] &amp; 1" represents bit 0, "<paramref name="values"/>[0] &amp; 2" represents bit 1,
+        /// "<paramref name="values"/>[0] &amp; 4" represents bit 2, and so on.
+        /// </remarks>
+        public BitArray(int[] values)
+        {
+            ArgumentNullException.ThrowIfNull(values);
+            if (values.Length > int.MaxValue / BitsPerInt32)
+            {
+                throw new ArgumentException(SR.Format(SR.Argument_ArrayTooLarge, BitsPerInt32), nameof(values));
+            }
+
+            _bitLength = values.Length * BitsPerInt32;
+            _array = AllocateByteArray(_bitLength);
+
+            if (BitConverter.IsLittleEndian)
+            {
+                MemoryMarshal.AsBytes(values.AsSpan()).CopyTo(_array);
```

Comment:
Does this need `values.AsSpan()`? Can't it just be `MemoryMarshal.AsBytes(values).CopyTo(_array)`?

Reply:
It can be removed. I've built up muscle memory I need to fight from working in libs recently with netstandard2.0 as a TFM.